classes/test.php

```diff
@@ -748,7 +749,7 @@ public function setTestNamespace($testNamespace)
 			throw new exceptions\logic\invalidArgument('Test namespace must not be empty');
 		}
 
-		if (!self::isRegex($testNamespace) && !self::isValidNamespace($testNamespace, true))
+		if (!analyzer::isRegex($testNamespace) && !analyzer::isValidNamespace($testNamespace, true))
```

Comment:
Using static calls here does not seem to be good: I think this is why we need to mock the `preg_match` function in some tests.

Using an instance of `analyzer` and injecting it in the `test` instance might be better for testability.